src/components/PeopleFilters.tsx

```diff
@@ -1,17 +1,152 @@
-export const PeopleFilters = () => {
+import React, { useCallback } from 'react';
+import classNames from 'classnames';
+import { NavLink, useSearchParams, useLocation } from 'react-router-dom';
+import { FEMALE, MALE } from '../utils/constants';
+
+interface Props {
+  setSearchQuery: React.Dispatch<React.SetStateAction<string>>;
+  setSex: React.Dispatch<React.SetStateAction<string>>;
+  setCenturies: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+const setActiveClass = ({ isActive }: { isActive: boolean }) => {
+  return classNames({ 'is-active': isActive });
+};
+
+export const PeopleFilters: React.FC<Props> = ({
+  setSearchQuery,
+  setSex,
+  setCenturies,
+}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const location = useLocation();
+
+  const isFemalePicked = location.search.includes('sex=f');
+  const isMalePicked = location.search.includes('sex=m');
+  const isSexPicked = isMalePicked || isFemalePicked;
+
+  const searchHandler = useCallback((
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const newSearchQuery = event.target.value;
+
+    const newSearchParams = new URLSearchParams(searchParams.toString());
+
+    newSearchParams.set('searchQuery', newSearchQuery);
+
+    if (newSearchQuery === '') {
+      newSearchParams.delete('searchQuery');
+    }
+
+    setSearchQuery(newSearchQuery);
+    setSearchParams(newSearchParams);
+  }, []);
```

Comment:
In the useCallback hook, you should specify the setSearchParams as a dependency since it is used within the callback.